floris/wind_data.py

```diff
@@ -482,48 +508,61 @@ def upsample(self, wd_step=None, ws_step=None, method="linear", inplace=False):
         else:
             value_matrix = None
 
-        # If the first entry of wind_direction column is 0, and the last entry is not 360, then
-        # pad 360 to the end of the wind direction column and the last row of the ti_matrix and
-        # freq_matrix by copying the 0 entry
-        if len(wind_direction_column) > 1:
-            if wind_direction_column[0] == 0 and wind_direction_column[-1] != 360:
-                wind_direction_column = np.append(wind_direction_column, 360)
-                ti_matrix = np.vstack((ti_matrix, ti_matrix[0, :]))
-                freq_matrix = np.vstack((freq_matrix, freq_matrix[0, :]))
-                if self.value_table is not None:
-                    value_matrix = np.vstack((value_matrix, value_matrix[0, :]))
-
-        # If the wind_direction columns has length 1, then pad the wind_direction column with
-        # that value + and - 1 and expand the matrices accordingly
-        # (this avoids interpolation errors)
-        if len(wind_direction_column) == 1:
-            wind_direction_column = np.array(
-                [
-                    wind_direction_column[0] - 1,
-                    wind_direction_column[0],
-                    wind_direction_column[0] + 1,
-                ]
-            )
-            ti_matrix = np.vstack((ti_matrix, ti_matrix[0, :], ti_matrix[0, :]))
-            freq_matrix = np.vstack((freq_matrix, freq_matrix[0, :], freq_matrix[0, :]))
-            if self.value_table is not None:
-                value_matrix = np.vstack((value_matrix, value_matrix[0, :], value_matrix[0, :]))
-
-        # If the wind_speed column has length 1, then pad the wind_speed column with
-        # that value + and - 1
-        # and expand the matrices accordingly (this avoids interpolation errors)
-        if len(wind_speed_column) == 1:
-            wind_speed_column = np.array(
-                [wind_speed_column[0] - 1, wind_speed_column[0], wind_speed_column[0] + 1]
-            )
-            ti_matrix = np.hstack((ti_matrix, ti_matrix[:, 0][:, None], ti_matrix[:, 0][:, None]))
-            freq_matrix = np.hstack(
-                (freq_matrix, freq_matrix[:, 0][:, None], freq_matrix[:, 0][:, None])
+        # Make sure everything sorted by wind direction and wind speed
+        # This should be the case but it's possible user can pass in unsorted data
```

Comment:
Again, the user could also pass in data with irregular spacing, but I don't think we currently can handle that?

Reply:
Fixed now